src/main.cpp

```diff
@@ -3368,16 +3368,16 @@ bool FindUndoPos(CValidationState &state, int nFile, CDiskBlockPos &pos, unsigne
     return true;
 }
 
-bool CheckBlockHeader(const CBlockHeader& block, CValidationState& state, const Consensus::Params& consensusParams, bool fCheckPOW)
+bool CheckBlockHeader(const CBlockHeader& block, CValidationState& state, const uint256& powLimit, bool fCheckPOW)
```

Comment:
Since you are touching CheckBlockHeader, perhaps you could use the opportunity to remove the param fCheckPOW.
Now checking the pow is the only thing CheckBlockHeader does, so if you don't want to check it just don't call the function.